### PR TITLE
Allow teams with empty Applications deletion

### PR DESCRIPTION
**Pull request by hardillb**
part of #2559

## Description

If only empty Appliations are present in the team then it can be deleted.

This is to reduce the friction for uses wanting to stop billing for teams.

## Related Issue(s)

#2559

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by hardillb** [comment]
Need to write some tests for this before merging.

**Comment by hardillb** [comment]
Also need to fix the UI tests

File: frontend/src/pages/team/Settings/Danger.vue [hunk 1]
```diff
@@ -62,13 +63,28 @@ export default {
...
-            return this.applicationCount === 0
+            if (this.applicationCount === 0) {
+                return true
+            } else {
+                for (let i = 0; i < this.applicationList.count; i++) {
+                    if (this.applicationList.applications[i].instances.length !== 0) {
+                        return false
+                    }
+                }
+                return true
```
Comment (Pezmc) [inline frontend/src/pages/team/Settings/Danger.vue:74]: this.applicationList.applications.some((application) => application.instances.length !== 0)
```suggestion
                return this.applicationList.applications.every((application) => application.instances.length === 0)
```

File: frontend/src/pages/team/Settings/Danger.vue [hunk 2]
```diff
@@ -62,13 +63,28 @@ export default {
...
-            if (this.applicationCount > 0) {
-                return 'You cannot delete a team that still owns applications.'
+            if (this.applicationCount === 0) {
+                return 'Deleting the team cannot be undone. Take care.'
             } else {
+                for (let i = 0; i < this.applicationList.count; i++) {
+                    if (this.applicationList.applications[i].instances.length !== 0) {
+                        return 'You cannot delete a team that still owns instances.'
+                    }
+                }
```
Comment (Pezmc) [inline frontend/src/pages/team/Settings/Danger.vue:85]:
```suggestion
                if (this.applicationList.applications.some((application) => application.instances.length !== 0) {
                    return 'You cannot delete a team that still owns instances.'
                }
```

File: frontend/src/pages/team/Settings/Danger.vue [hunk 3]
```diff
@@ -52,6 +52,7 @@ export default {
     },
     data () {
         return {
+            applicationList: {},
             applicationCount: -1,
```
Comment (Pezmc) [inline frontend/src/pages/team/Settings/Danger.vue:56]: Given we have `applicationList.count` now I'd drop `applicationCount` or make it a computed prop based on `applicationList` to save the duplication

File: frontend/src/pages/team/Settings/Danger.vue [hunk 4]
```diff
@@ -62,13 +63,28 @@ export default {
...
+            if (this.applicationCount === 0) {
+                return 'Deleting the team cannot be undone. Take care.'
             } else {
+                for (let i = 0; i < this.applicationList.count; i++) {
+                    if (this.applicationList.applications[i].instances.length !== 0) {
+                        return 'You cannot delete a team that still owns instances.'
+                    }
+                }
                 return 'Deleting the team cannot be undone. Take care.'
+                // return 'You cannot delete a team that still owns instances.'
```
Comment (Pezmc) [inline frontend/src/pages/team/Settings/Danger.vue:87]: ✂️

**Review by Pezmc — DISMISSED** [review]
Have suggested some ES6 magic, please validate it though!

**Review by knolleary — APPROVED** [review]

**Comment by knolleary** [comment]
@hardillb looks good - happy to take this out of draft and merge?

**Comment by hardillb** [comment]
Hi, yep all good

**Pull request merged**